### PR TITLE
add routes' http methods to allowed

### PR DESCRIPTION
**Pull request by koschkosch**
added some additional code in reg_routes to set methods given through routes automatically to $self->{allowed_methods}.

### CODE REVIEW

File: lib/AnyEvent/HTTPD/Router.pm [hunk 1]
```diff
@@ -47,14 +50,36 @@ sub dispatcher { shift->{dispatcher} }
 sub reg_routes {
     my $self = shift;
 
-    croak 'arguemnts to reg_routes are required' if @_ == 0;
+    croak 'arguments to reg_routes are required' if @_ == 0;
     croak 'arguments to reg_routes are confusing' if @_ % 3 != 0;
+    my $route_class = $self->{route_class};
```
Comment (ufobat) [inline lib/AnyEvent/HTTPD/Router.pm]: $route_class is from a old version of this module and not needed, could you remove line 55

File: lib/AnyEvent/HTTPD/Router.pm [hunk 2]
```diff
@@ -14,6 +14,9 @@ sub new {
     my $class = ref($this) || $this;
     my %args  = @_;
 
+	# set allowed methods to GET until we get some routes
+    # why GET? because :verbs will need at least one real HTTP method
+    $args{allowed_methods} = ['GET'] unless $args{allowed_methods};
```
Comment (ufobat) [inline lib/AnyEvent/HTTPD/Router.pm]: I would take (or keep the defaults from super-class) that AE::HTTPD has in it's module. (which includes GET, and some more)
so you could leave out this line as well

**Review by ufobat — CHANGES_REQUESTED** [review]
thanks :-) could you please have a look at the 2 comments.

File: lib/AnyEvent/HTTPD/Router.pm (hunk 1 above)
Comment (koschkosch) [reply lib/AnyEvent/HTTPD/Router.pm]: fixed. Thanks

File: lib/AnyEvent/HTTPD/Router.pm (hunk 2 above)
Comment (koschkosch) [reply lib/AnyEvent/HTTPD/Router.pm]: Removed. Thanks

**Pull request merged**